core/trino-main/src/main/java/io/trino/execution/scheduler/faulttolerant/BinPackingNodeAllocatorService.java

```diff
@@ -755,10 +768,14 @@ private List<InternalNode> dropCoordinatorsIfNecessary(List<InternalNode> candid
         public ReserveResult tryReserve(PendingAcquire acquire)
         {
             NodeRequirements requirements = acquire.getNodeRequirements();
-            Optional<Set<InternalNode>> catalogNodes = requirements.getCatalogHandle().map(nodesSnapshot::getConnectorNodes);
 
-            List<InternalNode> candidates = new ArrayList<>(allNodesSorted);
```

Comment:
This could be just:
```
         return requirements.getCatalogHandle()
                    .map(catalogNodes::getUnchecked)
                    .orElse(workerNodesSorted);
```

Reply:
changed but it is getting more complex later anuyway